packages/graphql-server/src/functions/__tests__/useRequireAuth.test.ts

```diff
@@ -0,0 +1,352 @@
+
+    const handlerEnrichedWithAuthentication = useRequireAuth({
+      handlerFn: handler,
+      getCurrentUser,
+    })
+
+    const headers = {
+      'auth-provider': 'custom',
+      authorization: 'Bearer myToken',
+    }
+
+    const output = await handlerEnrichedWithAuthentication(
+      mockedAuthenticationEvent({ headers }),
+      {}
+    )
+
+    const resBody = JSON.parse(output.body)
+    expect(resBody.token).toEqual('myToken')
+  })
+
+  it('Updates context with output of current user with roles', async () => {
+    // @MARK
+    // Because we use context inside useRequireAuth, we only want to import this function
+    // once we disable context isolation for our test
+    const { useRequireAuth } = require('../useRequireAuth')
+
+    const handlerEnrichedWithAuthentication = useRequireAuth({
+      handlerFn: handler,
+      getCurrentUser,
+    })
+
+    // The authorization JWT is valid and has roles in app metadata
+    // {
+    //   "sub": "1234567891",
+    //   "name": "John Editor",
+    //   "iat": 1516239022,
+    //   "app_metadata": {
+    //      "roles": ["editor"]
+    //   }
+    // }
+    const headersWithRoles = {
+      'auth-provider': 'netlify',
+      authorization:
+        'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkxIiwibmFtZSI6IkpvaG4gRWRpdG9yIiwiaWF0IjoxNTE2MjM5MDIyLCJhcHBfbWV0YWRhdGEiOnsicm9sZXMiOlsiZWRpdG9yIl19fQ.Fhxe58-7BcjJDoYQAZluJYGwPTPLU0x6K5yA3zXKaX8',
+    }
+
+    const output = await handlerEnrichedWithAuthentication(
+      mockedAuthenticationEvent({ headers: headersWithRoles }),
+      {}
+    ) // ?
+
+    const resBody = JSON.parse(output.body)
+    expect(resBody.name).toEqual('John Editor')
+    expect(resBody.roles).toContain('editor')
+  })
+  it('is 401 Unauthenticated status if an error occurs when getting current user info', async () => {
+    const { useRequireAuth } = require('../useRequireAuth')
+
+    const handlerEnrichedWithAuthentication = useRequireAuth({
+      handlerFn: handler,
+      getCurrentUser: getCurrentUserWithError,
+    })
+
+    const customHeaders = {
+      'auth-provider': 'custom',
+      authorization: 'Bearer myToken',
+    }
+
+    const response = await handlerEnrichedWithAuthentication(
+      mockedAuthenticationEvent({ headers: customHeaders }),
+      {}
+    )
+
+    expect(response.statusCode).toEqual(401)
+  })
+
+  it('is 401 Unauthenticated status if no auth headers present', async () => {
+    const { useRequireAuth } = require('../useRequireAuth')
+
+    const handlerEnrichedWithAuthentication = useRequireAuth({
+      handlerFn: handler,
+      getCurrentUser,
+    })
+
+    const missingHeaders = null
+
+    const response = await handlerEnrichedWithAuthentication(
+      mockedAuthenticationEvent({ headers: missingHeaders }),
+      {}
+    )
+
+    expect(response.statusCode).toEqual(401)
+  })
+
+  it('is 401 Unauthenticated status if the auth provider is unsupported', async () => {
```

Comment:
I don't think it should do this! unsupported auth providers should just return the token (like custom)

Reply:
Actually it does work as you said the auth exception is because I don’t set a authorization header. So, I can just remove the test

Reply:
Changing test to return token if provider is unsupported